gpt4all-chat/src/toolcallparser.cpp

```diff
@@ -59,34 +92,59 @@ void ToolCallParser::update(const QString &update)
         case ToolEnums::ParseState::None:
             {
                 m_expectedIndex = 1;
-                m_expected = ToolCallStart.at(1);
-                m_state = ToolEnums::ParseState::InStart;
+                setExpected(m_possibleStartTags);
+                m_state = ToolEnums::ParseState::InTagChoice;
                 m_startIndex = i;
                 break;
             }
+        case ToolEnums::ParseState::InTagChoice:
+            {
+                // WARNING: this assumes the first char of the tags differ
```

Comment:
If this is a FIXME, it should be `FIXME(adam)`. If this is a note, there should be a `NOTE: ` or `NB: ` (nota bene) near the constants in toolcallparser.h.

Maybe both?